bosh-cf-crossregion/azuredeploy.json

```diff
@@ -0,0 +1,813 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmName": {
+      "type": "string",
+      "metadata": {
+        "description": "name of the vm, will be used as DNS Name for the Public IP used to access the Virtual Machine"
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "User name for the Virtual Machine"
+      }
+    },
+    "sshKeyData": {
+      "type": "string",
+      "metadata": {
+        "description": "Please copy the content of your SSH RSA public key and paste it here. You can use \"ssh-keygen -t rsa -b 2048\" to generate your SSH key pairs."
+      }
+    },
+    "environment": {
+      "type": "string",
+      "allowedValues": [
+        "AzureCloud",
+        "AzureChinaCloud"
+      ],
+      "metadata": {
+        "description": "Different environments in Azure. Choose AzureCloud for Global Azure, and choose AzureChinaCloud for Mooncake (Azure China Cloud)."
+      }
+    },
+    "tenantID": {
+      "type": "string",
+      "minLength": 36,
+      "maxLength": 36,
+      "metadata": {
+        "description": "ID of the tenant. See https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/blob/master/docs/guidance.md"
+      }
+    },
+    "clientID": {
+      "type": "string",
+      "minLength": 36,
+      "maxLength": 36,
+      "metadata": {
+        "description": "ID of the client. See https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/blob/master/docs/guidance.md"
+      }
+    },
+    "clientSecret": {
+      "type": "securestring",
+      "metadata": {
+        "description": "secret of the client. See https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/blob/master/docs/guidance.md"
+      }
+    },
+    "autoDeployBosh": {
+      "type": "string",
+      "defaultValue": "enabled",
+      "allowedValues": [
+        "enabled",
+        "disabled"
+      ],
+      "metadata": {
+        "description": "The flag that enables automatic deployment of Bosh"
+      }
+    },
+    "primaryLocation": {
+      "type": "string",
+      "allowedValues": [
+          "North Europe",
+          "West Europe",
+          "East US",
+          "West US"
+      ],
+      "metadata": {
+        "description": "The primary location to provision resources for Bosh and CF"
+      }
+    },
+    "secondaryLocation": {
+      "type": "string",
+      "allowedValues": [
+          "North Europe",
+          "West Europe",
+          "East US",
+          "West US"
+      ],
+      "metadata": {
+        "description": "The secondary location to provision resources for Bosh and CF"
+      }
+    },
```

Comment:
@bingosummer 
1. Does not need to be paired-locations, although the closer the better since bosh director needs to communicate with secondary nodes to deploy.
2. These are four locations tested with support for both virtualNetworkGateways and premium storage accounts.
3. Based on my test results, it's recommended to create the resource group in the same location as the `primaryLocation` for the primary nodes while the `secondaryLocation` should be close to the `primaryLocation` to enhance performance.